### PR TITLE
Doc update for Utf8Encoder and Utf8Decoder

### PR DESCRIPTION
**Pull request by jpsiitonen**
Code examples added for Utf8Encoder and Utf8Decoder

### CODE REVIEW

File: sdk/lib/convert/utf.dart [hunk 1]
```diff
@@ -74,6 +74,14 @@ class Utf8Codec extends Encoding {
 
 /// This class converts strings to their UTF-8 code units (a list of
 /// unsigned 8-bit integers).
+///
+/// Example:
+/// ```dart
+/// const String sample = '"Îñţérñåţîöñåļîžåţîờñ"';
+/// const Utf8Encoder utf8encoder = Utf8Encoder();
```
Comment (alexlindroos) [inline sdk/lib/convert/utf.dart]: utf8Encoder

File: sdk/lib/convert/utf.dart [hunk 2]
```diff
@@ -74,6 +74,14 @@ class Utf8Codec extends Encoding {
...
+/// final List<int> encoded = utf8encoder.convert(sample);
```
Comment (alexlindroos) [inline sdk/lib/convert/utf.dart]: We have been using `encoded`and `encodedSample` in some of the examples. We should stick with one style

File: sdk/lib/convert/utf.dart [hunk 3]
```diff
@@ -285,6 +293,32 @@ class _Utf8EncoderSink extends _Utf8Encoder with StringConversionSinkMixin {
 
 /// This class converts UTF-8 code units (lists of unsigned 8-bit integers)
 /// to a string.
+///
+/// Example:
+/// ```dart
+/// const Utf8Decoder utf8decoder = Utf8Decoder();
```
Comment (alexlindroos) [inline sdk/lib/convert/utf.dart]: utf8Decoder

File: sdk/lib/convert/utf.dart [hunk 4]
```diff
@@ -285,6 +293,32 @@ class _Utf8EncoderSink extends _Utf8Encoder with StringConversionSinkMixin {
...
+/// const List<int> encodedBytes =
+/// [195, 142, 195, 177, 197, 163, 195, 169, 114, 195, 177, 195, 165, 197,
+///  163, 195, 174, 195, 182, 195, 177, 195, 165, 196, 188, 195, 174, 197,
+///  190, 195, 165, 197, 163, 195, 174, 225, 187, 157, 195, 177];
+///
+/// final String decoded = utf8decoder.convert(encodedBytes);
```
Comment (alexlindroos) [inline sdk/lib/convert/utf.dart]: decoded or decodedBytes?

File: sdk/lib/convert/utf.dart [hunk 5]
```diff
@@ -285,6 +293,32 @@ class _Utf8EncoderSink extends _Utf8Encoder with StringConversionSinkMixin {
...
+/// ```
+/// If encoded data contains invalid values, the decoder
+/// throws a [FormatException]
+///
+/// If `allowMalformed` is set to true,
+/// the decoder replaces the invalid bytes with the character `U+FFFD` �.
+///
+/// Allow invalid byte value example:
+/// ```dart
+/// const Utf8Decoder utf8decoder = Utf8Decoder(allowMalformed: true);
```
Comment (alexlindroos) [inline sdk/lib/convert/utf.dart]: utf8Decoder

**Review by tomgilder — APPROVED** [review]

**Review by alexlindroos — APPROVED** [review]

**Review by stepthomp — APPROVED** [review]

**Pull request closed without merging**